File: src/game/characters/Gohan.ts
import { BaseFighter } from "../BaseFighter";

export class Gohan extends BaseFighter {
  readonly id = "gohan" as const;
  readonly name = "Gohan";
  readonly greeting = "I won't let you hurt anyone.";
  readonly signatureAttack = "Masenko";
  readonly attackStyle = "orb" as const;
  readonly power = 10000;
  readonly speed = 86;
  readonly defense = 76;
  readonly criticalChance = 0.22;
  readonly baseDamage = 21;
  readonly nativeFacing = "right" as const;
  readonly scale = 0.98;
  readonly palette = {
    skin: "#efae7a",
    hair: "#17171c",
    primary: "#5d2b8c",
    secondary: "#32225f",
    accent: "#eecf4c",
    aura: "#ffe45a"
  };
}

File: csharp-reference/Characters/Gohan.cs
using DragonBallArena.Models;

namespace DragonBallArena.Characters;

public sealed class Gohan(string? name = null) : BaseFighter(name)
{
    public override string Id => "gohan";
    public override string DisplayName => "Gohan";
    public override string Greeting => "I won't let you hurt anyone.";
    public override string SignatureAttack => "Masenko";
    public override string AttackStyle => "orb";
    public override double Scale => 0.98;
    public override FighterStats Stats => new(10000, 86, 76, 0.22, 21);
    public override FighterPalette Palette => new(
        "#efae7a", "#17171c", "#5d2b8c", "#32225f", "#eecf4c", "#ffe45a");
}
